lib/flight_scheduler/message_processor.rb

```diff
@@ -30,6 +30,9 @@ module FlightScheduler
   # Process incoming messages and send responses.
   #
   class MessageProcessor
+    # TODO: Remove the particular instance of connection as it might drop out
+    # during long running operations. Instead use MessageSender which polls for
+    # the currently open connection
```

Comment:
Good idea.  I like `MessageSender`.  We can make that change piecemeal as we edit `MessageProcessor`.  (Not that I'd object to you making the change as part of this PR :-).

Reply:
I've made a wholesale change to `MessageSender` except the `CONNECTED` message. It is sent after the connection is established and probably shouldn't be preformed asynchronously

https://github.com/openflighthpc/flight-scheduler-daemon/pull/16/commits/76f0fc34234e63727d7e7c7a98de8fc3ad1b186d